docs/index.md

```diff
@@ -4,60 +4,67 @@ id: Overview
 
-In Old Norse, Old High German and Old English a "lind" is a shield constructed with two layers of linden wood. Linden wood shields are lightweight, and do not split easily, an appropriate metaphor for a sandboxing system which employs two technologies.
+In Old Norse, Old High German and Old English a "lind" is a shield constructed with two layers of linden wood. Linden wood shields are lightweight, and do not split easily, an appropriate metaphor for a sandboxing system which is lightweight and which provides layered security.
 
 ## Core Concepts
 
-- **Cage**: Lightweight isolation boundary within a process
-    - Can run legacy code (may need recompilation)
-    - Protects and isolates memory
-- **Microvisor**: Small POSIX compliant kernel within a process
-    - Provides a POSIX interface
-    - Distinct isolation between cages
-- **3i (three eye)**: Capability-based POSIX interfaces between cages
+- **Cage**: This term describes the isolated memory namespace that an application executes in.  It is analogous to a process in Linux. 
+    - Can run legacy code compiled with Wasm as a target
+    - Protects and isolates memory, control flow, etc.
+- **Microvisor**: RawPOSIX is a small kernel running within the Lind process.  This is analogous to the Linux kernel.  Note, however, that this and all of the rest of Lind runs as an unprivileged Linux process.
+    - Provides a POSIXish interface (runs most Linux programs)
+    - Handles file descriptor separation, fork, exec, signals, threading, etc.
+- **3i (three eye)**: Capability-based POSIX interface to call between cages or into the microvisor.  This is conceptually similar to a programmable system call table and IPC interface. 
+    - Each cage has a separate system call table which can be independently changed to redirect into other cages or the microvisor
+    - Fast, isolated calling between cages
+    - Enables complex functionality (system call filtering, file systems, proxies, etc.) to be external to the microvisor
 
 ## Technology Overview
 
 ### Cages
-Memory and bookkeeping that encapsulates the idea of a typical OS process, encompassing applications as well as grates.
+A cage is simply a Linux process.  The code must recompiled to run in Wasm and linked to our modified glibc so that it makes its system calls through 3i.  However, the source code for applications only needs to be modified in rare cases (such as applications that directly make system calls).
```

Comment:
```suggestion
A cage runs a Linux application in an isolated part of the Lind process's namespace.  The code must recompiled to run in Wasm and linked to our modified glibc so that it makes its system calls through 3i.  However, the source code for applications only needs to be modified in rare cases (such as applications that directly make system calls).
```